src/validation.cpp

```diff
@@ -132,6 +132,10 @@ CTxMemPool mempool(&feeEstimator);
 /** Constant stuff for coinbase transactions we create: */
 CScript COINBASE_FLAGS;
 
+CTxMemPool* MempoolInstance()
+{
+    return &::mempool;
+}
```

Comment:
If this function is supposed to replace uses of ::mempool in existing code, I think it'd be better to return a reference:

```c++
CTxMemPool& MempoolInstance()
{
    return ::mempool;
}
```

now and change it to:

```c++
CTxMemPool& MempoolInstance()
{
    CHECK_NONFATAL(::g_mempool);
    return *::g_mempool;
}
```

in the future when mempool is replaced by an optional or unique_ptr or similar.

Using a reference instead of a pointer for existing code is better because it won't burden the code with checking for nulls.

re: https://github.com/bitcoin/bitcoin/pull/17407/files#issue-338259412

> Finally, in the future someone might want to run a consensus-only full node"

If new code is going to be added or existing code is going to be changed that expects the mempool to be null, this code could probably just access `g_mempool` directly instead of using the helper function. Or you could add different helper functions `CTxMemPool* OptionalMempoolInstance()` or `bool MempoolEnabled()` if referencing the global is a problem.

Reply:
> If this function is supposed to replace uses of ::mempool in existing code, I think it'd be better to return a reference:

It is *not* supposed to replace ::mempool in existing code. It is supposed to be what you refer to as "different helper function". I am happy to rename it to `OptionalMempoolInstance`, but I think that that is a bit too verbose.

Reply:
Unclear what the next steps are here. It's hard to see how a function that doesn't log or check anything but just returns a pointer to a global could be useful, but it'd help if you could describe more literally how the function and global and function calls will change in the future.